storage/object/api-cli/combining-iam-and-object-storage.mdx

```diff
@@ -0,0 +1,273 @@
+tags: object storage command bucket s3 iam permissions acl policy
+dates:
+  validation: 2023-01-23
+  posted: 2023-01-17
+---
+
+<Macro id="iam-requirements" />
+
+<Message type="requirement">
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have [configured your API keys](/identity-and-access-management/iam/how-to/create-api-keys/)
+  - You have [created an Object Storage bucket](/storage/object/how-to/create-a-bucket/)
+  </Message>
+
+Identity and Access Management (IAM) brings a set of new features to your Scaleway account, allowing you to control and manage access to your resources. For users of Scaleway Object Storage, this means that besides configuring access via [bucket policies](/storage/object/api-cli/bucket-policy/), you can now use IAM policies to configure permissions.
+
+An IAM policy is used to define the permissions of users, groups and applications in a given Organization. A policy is composed of a principal (the user, group or application to which it applies) and one or more IAM rules (which describe the permission sets the principal should have, and the scope of those permission sets).
+
+Bucket policies are a resource-based S3 policy option that allow you to grant users and/or applications access to a specific bucket in a project. The different component strings of a bucket policy allow you to configure fine-grained permissions.
+
+Bucket policies behave like objects: they can be uploaded into buckets. Once you upload, or "put", a bucket policy into a bucket, they take immediate effect and will from then on define who can access and perform actions on the bucket.
+
+## Defining whether to use IAM or bucket policies
+
+IAM can be used when you do not need to configure project-level access to specific buckets, and favor simplicity over granularity. IAM is central, and policy-based: it says where people can perform actions in the entire Scaleway ecosystem. For example, if access to resources at project level is enough for your Object Storage use cases, we advise you to use IAM only.
+
+Bucket policies operate at the bucket level, and enable you to state clearly who can perform different actions on which specific bucket.
+
+If you need to configure more granular access to your Object Storage buckets, you can combine both IAM and bucket policies. For example, if you must restrict access to specific actions you can perform on a bucket, or if you need the bucket to be available only to certain users, you can use a combination of IAM and bucket policies.
+
+## Combining IAM and bucket policies
+
+In order to perform actions on a bucket, a principal (user or application) must be granted access via an IAM policy, and optionally via a bucket policy.
+
+For example, if a user wants to list the objects in a bucket, they must ensure that:
```

Comment:
```suggestion
For example, if a user wants to list the objects in a bucket, he must ensure that:
```

or 


```suggestion
For example, if users want to list the objects in a bucket, they must ensure that:
```

Reply:
"they" can be used as the third person singular in some cases. In this case, it's especially used to indicate a generic individual and ensure gender neutrality :) 

you can read more about it here: https://en.wikipedia.org/wiki/Singular_they

Reply:
Ohh sorry I didn't know that. 😅 Thank you for the link! 🙂